.github/workflows/release.yml

```diff
@@ -77,17 +77,55 @@ jobs:
           service_account_key: ${{ secrets.GCP_SA_KEY }}
           export_default_credentials: true
       - name: Upload zip
-        uses: google-github-actions/upload-cloud-storage@v0
+        uses: google-github-actions/upload-cloud-storage@v1
         with:
           path: ${{ env.OUTNAME }}.zip
           destination: ${{ secrets.GCP_BUCKET }}/${{ github.ref_name }}/
 
   release:
     runs-on: ubuntu-latest
     needs: build-and-upload
+    
     steps:
-      - name: Check out Git repository
-        uses: actions/checkout@v4
+      - name: Generate the files names
+        shell: bash
+        run: |
+          echo "OUTNAME_WIN_AMD64=go-spacemesh-${{ github.ref_name }}-win-amd64" >> $GITHUB_ENV
+          echo "OUTNAME_LINUX_AMD64=go-spacemesh-${{ github.ref_name }}-linux-amd64" >> $GITHUB_ENV
+          echo "OUTNAME_LINUX_ARM64=go-spacemesh-${{ github.ref_name }}-linux-arm64" >> $GITHUB_ENV
+          echo "OUTNAME_MAC_AMD64=go-spacemesh-${{ github.ref_name }}-mac-amd64" >> $GITHUB_ENV
+          echo "OUTNAME_MAC_ARM64=go-spacemesh-${{ github.ref_name }}-mac-arm64" >> $GITHUB_ENV
+
+      - name: Calculate hashsums of release files
+        shell: bash
+        run: |
+          wget https://storage.googleapis.com/${{ secrets.GCP_BUCKET }}/${{ github.ref_name }}/${{ env.OUTNAME_WIN_AMD64 }}.zip
```

Comment:
We should really not re-download files. if the attacker has access to the bucket then he could overwrite them on the fly.

Reply:
Fixed @pigmej